src/lib/icu/meson.build

```diff
@@ -18,8 +18,18 @@ if icu_dep.found()
     'Init.cxx',
   ]
 elif not get_option('iconv').disabled()
-  have_iconv = compiler.has_function('iconv', prefix : '#include <iconv.h>')
-  conf.set('HAVE_ICONV', have_iconv)
+  have_iconv = c_compiler.has_header_symbol('iconv.h', 'iconv')
```

Comment:
Why switch from `has_function` to `has_header_symbol`?

Reply:
has_function was moved down.

Reply:
OK, `has_function` is still there, but that doesn't answer why you used `has_header_symbol` now.
Also, your change uses `libiconv` even if the C library already provides `iconv()`. Why did you decide to do that?

Reply:
libiconv usually has more character maps. In OpenWrt specifically, some of those maps in musl are patched out.

Reply:
If you prefer the external libiconv, there should be a comment in the Meson source explaining it.
This change was not mentioned in your commit message, so finding this as surprising for me.